### ISSUE TITLE
Android app crashes after naming device

### ISSUE THREAD

**Issue by ricklupton** [keybase/client#6993]
Android version 4.2.2
Keybase version 1.0.0-alpha-2

I signed into my Keybase account using my username and my desktop device. Adding the code worked ok. I named the device and clicked continue, then the app crashes.

If I reopen it, I can see that the "Phone" device has been created (it's in the list of devices to use for signing in, along with the desktop and the paper key) but it doesn't seem to be associated with the app.4.2.2.

I revoked the device and tried again, but the same thing happened.

**Comment by techmagus** [comment]
Experiencing the same. Works fine with Android M but not Jellybean.

Maybe Jellybean support can be dropped altogether, and concentrate on KK, M, N, and Android O? (My Jellybean phone is only serving as secondary/backup.)

**Comment by maxtaco** [comment]
@techmagus can you push the Gear > "Send Feedback" and send up your logs? We can take a look, thanks!

**Comment by ricklupton** [comment]
@maxtaco I can't get as far as seeing a gear icon -- the options are just "create an account" and "log in". As I described above, I tried logging in but the app crashes. Is there another way of sending the logs?

(I'd definitely like to see Jellybean support kept as I'm still using it!)

**Comment by maxtaco** [comment]
cc @cjb

**Comment by cjb** [comment]
Perhaps best for us to just try to repro the bug ourselves in a Jellybean simulator, I'll add an internal bug report to do that.

(Haven't reached an opinion yet on how much effort we should put in to supporting Jellybean.)

**Comment by berkes** [comment]
Any progress on this? Anything we can do to help?

**Comment by specious** [comment]
Same here. Android app crashed immediately after entering text verification code and now won't let me log in even though the device is listed as verified.

Android 4.2.2 on GT-I8200.

What can I do to help debug this?

![977f830e-1e5a-4fa9-a275-0ef7b0bee4a8](https://user-images.githubusercontent.com/451178/34082220-4a03653a-e328-11e7-87d7-7601c89e4dad.jpeg)
![bbaa4214-cd57-458e-be77-f76c39de3092](https://user-images.githubusercontent.com/451178/34082295-48c07c84-e329-11e7-8d1c-3c7d02084faa.jpeg)
![1fb156ce-3f30-4bb8-8fff-0220de34daba](https://user-images.githubusercontent.com/451178/34082226-5bce5a18-e328-11e7-8958-c0fff230ffcc.png)

**Comment by specious** [comment]
Submitted report from the Android.

![cca7743d-a033-4044-bee4-f5563b714103](https://user-images.githubusercontent.com/451178/34082340-26121494-e32a-11e7-9a9a-159503d572a7.jpeg)

**Comment by maxtaco** [comment]
@specious I don't think we can support that version of Android, I think it's too old.  I hope the keybase app didn't cause your screen to crack BTW!

**Comment by specious** [comment]
@maxtaco, I absolutely concur that this Android is too old to officially bother supporting. Isn’t it curious though, that it’s possible to have a device that is listed as vefiried and yet it cannot log in? How is that happening?

**Comment by specious** [comment]
@maxtaco, I absolutely concur that this Android is too old to officially bother supporting. Isn’t it curious though, that it’s possible to have a device that is listed as verified and yet it cannot log in? How is that happening?

**Comment by maxtaco** [comment]
My guess is that the app crashed when trying to store the secret key to the Android keystore, and now the app can't read the private key back out.  So it got most the way through the provisioning process, but died when trying to access the key material.

**Comment by berkes** [comment]
@maxtaco and @specious It makes sense to not support this version of Android. Especially since it essentially is not supported right now. 

But in that case it should be limited on the play-store/f-droid and/or the AndroindManifest.xml.

That would keep people (like me) with an old device from trying to install it, and getting a crash. Which is far more annoying than seeing a "this app is not compatible with your device", which I am kinda used to, having such an old Android.

**Comment by maxtaco** [comment]
Good point. cc: @keybase/react-hackers

**Comment by jbonhag** [comment]
Not to mention the fact that I am eating up all these great device names, and now that they're revoked I can't re-use them 😆 😢

**Comment by maxtaco** [comment]
@jbonhag can you send feedback from the app?

**Comment by jbonhag** [comment]
Just sent the logs, hope they help!
